signer/core/signed_data.go

```diff
@@ -369,6 +369,7 @@ func (typedData *TypedData) HashStruct(primaryType string, data TypedDataMessage
 
 // Dependencies returns an array of custom types ordered by their hierarchical reference tree
 func (typedData *TypedData) Dependencies(primaryType string, found []string) []string {
+	primaryType = strings.TrimSuffix(primaryType, "[]")
```

Comment:
Isn't this a bit naive? 
If we use this approach, then it only solves `Foobar[]`, but not `Foobar[8]` or `Foobar[][]`. 
I'm not saying it's optimal as it is right now, but I'm kind of wondering how far to go with this.

Reply:
`Foobar[8]` is a data, which specific to a item in type `Foobar[]`, here `primaryType` means a type not a data

In my mind, the EIP-712 does not support 2d (or more) arrays?